esp-hal/src/i2c/master/mod.rs

```diff
@@ -85,6 +86,107 @@ const I2C_CHUNK_SIZE: usize = 254;
 // on ESP32 there is a chance to get trapped in `wait_for_completion` forever
 const MAX_ITERATIONS: u32 = 1_000_000;
 
+/// Address mode.
+///
+/// Note: This trait is sealed and should not be implemented outside of this
+/// crate.
+pub trait AddressMode: _private::AddressModeInternal + Copy + 'static {}
+
+mod _private {
+    pub trait AddressModeInternal {
+        fn as_i2c_address(&self) -> super::I2cAddress;
+    }
+
+    /// Internally used representation of an I2C address.
+    #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, strum::Display)]
+    #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+    pub enum I2cAddress {
+        /// Seven bit address
+        SevenBit(super::SevenBitAddress),
+    }
+
+    impl I2cAddress {
+        pub fn as_u8(&self) -> u8 {
```

Comment:
Why is this public, and what will happen when we introduce 10-bit addresses?

Reply:
can be pub(super) but since it's in a private module doesn't make much difference

10-bit address support will introduce some more changes - we'll probably have to match on the enum and do a few things differently (and will remove this function)